lightworks: 14.0.0 -> 14.5.0

gefla opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Lightworks is currently broken, see https://github.com/NixOS/nixpkgs/issues/80535

This updates to the most recent stable version and fixes the broken dependencies.

Note: Still doesn't fix all the issues - current issue appears to be patchelf related corruption:

```
 % ./result/bin/lightworks-14.5.0
Inconsistency detected by ld.so: dl-version.c: 205: _dl_check_map_versions: Assertion `needed != NULL' failed!
```

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Closes #80535

Commit:
Add libGL dependency

https://github.com/NixOS/nixpkgs/issues/80535

Commit:
Update Lightworks 14.0.0 -> 14.5.0

Commit:
Replace unavailable libjpeg.so.8 dependency

Commit:
Add/patch more missing dependencies

Pablo1107 committed:
ledger2beancount: init at 2.1

deviant committed:
iosevka-bin: add support for variants

This adds the ability to select a specific prebuilt variant. It also adds
an updater script for generating their hashes. Additionally, switching
to TTC files reduces the package size by an order of magnitude.

Example usage:

	fonts.fonts = with pkgs; [
	  (iosevka-bin.override { variant = "ss10"; })
	  (iosevka-bin.override { variant = "sparkle"; })
	  (iosevka-bin.override { variant = "aile"; })
	];

danderson committed:
nixos/tailscale: add tailscale to environment.systemPackages.

Use of Tailscale requires using the `tailscale` CLI to talk to the
daemon. If the CLI isn't in systemPackages, the resulting user experience
is confusing as the Tailscale daemon does nothing.

Signed-off-by: David Anderson <dave@natulte.net>

WhittlesJr committed:
bacnet-stack: init at 1.0.0

Update pkgs/tools/networking/bacnet-stack/default.nix

Co-authored-by: Daniel Løvbrøtte Olsen <daniel.olsen99+GitHub@gmail.com>

bacnet-stack: Add maintainer and use the original GitHub repo

bacnet-stack: Alphabetize

Update pkgs/top-level/all-packages.nix

Co-authored-by: Jon <jonringer@users.noreply.github.com>

alevy commented:
Confirming that these changes both avert the libGL issue and results in the error in the description

ericdallo committed:
flutter: Add dart cache to flutter

unode committed:
samtools: 1.10 -> 1.11

jonringer committed:
python3Packages.tensorflow: break if gast not correct

S-NA committed:
firefox: place alsaLib in lib search patch

libcubeb has dlopened libraries for awhile now. In nixpkgs there was
support for the PulseAudio backend doing this, however the ALSA backend
support was missed and caused issue #79310 (no sound with ALSA). This
gives ALSA users the ability to hear sound once again.

eyJhb committed:
among-sus: init at 2020-10-29

vbgl committed:
ocamlPackages.re: use released source rather than a git snapshot

This fixes the version information in the META file

vbgl committed:
ocamlPackages.ocp-build: 1.99.19-beta → 1.99.21-beta

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-26-gcfec46d from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/950856114c6c691a0b33763d8ee677e5361923a6.

sternenseemann committed:
haskellPackages.flat: unbreak

flat builds fine again as is.

mweinelt committed:
pythonPackages.spotipy: 2.16.0 -> 2.16.1

mweinelt committed:
pythonPackages.spotipy: disable tests and cleanup

Tests require network access.

jonringer committed:
scripts/mark-broken: improve

This improves on the previous verison of this script.

Previously it only accepted one attr, and required
explicit passing of the "broken" value.

This script is meant to be used to mark failing hydra builds as broken in the meta attrs
To use the script, you should pass the list of failing attrs as arguments to the script.

Example: `cat failing-attrs | xargs ./pkgs/common-update/scripts/mark-broken`

Generating a list of failing attrs: (this should be improved at a later date)
  - Go to the most recent hydra evaluation with all builds completed
  - Select the "builds still failing" tab
  - Highlight and select all packages, should be prefixed with `nixpkgs.`
  - Use regex and editor foo to leave only the attr names
  - Use the above example command to then execute the script

OTHER NOTES:
  - The `denyFileList` and `denyAttrList` will likely need to be updated slightly
    to align with the conventions used in nixpkgs at execution time
  - Any attrs which failed for any reason will be written to `failed-marks.txt`.
    Those attrs will likely need manual attention as disablement will likely be conditional.

zakame committed:
perlPackages.NetAsyncHTTP: init at 0.47

peti committed:
pandoc: update overrides for the new release

peti committed:
LTS Haskell 16.20

zakame committed:
perlPackages.NetAsyncWebSocket: init at 0.13

peterhoeg committed:
kdeApplications: 20.08.1 -> 20.08.2

peterhoeg committed:
okular: add missing dependency

peterhoeg committed:
kdeconnect: part of kdeApplications

peterhoeg committed:
zanshin: broken before the 20.08.2 upgrade

vbgl committed:
Revert "coq: propagate and install ocaml and findlib"

This reverts commit 5d0e2dedd5594cdf3ebbda8bc61310e227235a9c.

mweinelt committed:
homeassistant: 0.116.4 -> 0.117.0

mweinelt committed:
python3Packages.httpcore: 0.10.2 -> 0.12.0

Also clean up the package and enable tests.

grahamc committed:
make-disk-image: support legacy+gpt

grahamc committed:
nixos.amazonAmi: use legacy+gpt disk images to support partitions >2T

mweinelt committed:
python3Packages.httpx: 0.14.2 -> 0.16.1

Clean up package and convert tests to pytestCheckHook.

jonringer committed:
buildFHSUserEnvBubblewrap: fix whitespace with root directories

braunse committed:
aria2: refactor to avoid python2 dependency

Use python3 Sphinx package instead of python2 Sphinx package,
to avoid build-time dependency on python 2, as per
https://github.com/NixOS/nixpkgs/issues/101964

peterhoeg committed:
kdeconnect: avoid double-wrapping the binary

vbgl committed:
ocamlPackages.bistro: use Dune 2

vbgl committed:
ocamlPackages.webmachine: use Dune 2

vbgl committed:
ocamlPackages.cohttp: use Dune 2

vbgl committed:
ocamlPackages.jwto: use Dune 2

vbgl committed:
ocamlPackages.biocaml: use Dune 2

vbgl committed:
ocamlPackages.base64: 3.2.0 → 3.4.0

r-ryantm committed:
python37Packages.bids-validator: 1.5.6 -> 1.5.7

r-ryantm committed:
python37Packages.py_stringmatching: 0.4.1 -> 0.4.2

jonringer committed:
linux: add flavor metadata

jonringer committed:
linuxPackages.usbip: disable < 4.10

braunse committed:
rmlint: Use Python3 sphinx for docs

Eliminates a python2 dependency.

bbigras committed:
zenith: 0.10.1 -> 0.11.0

marsam committed:
Merge pull request #102008 from evenbrenden/jotta-cli-bump-version

jotta-cli: 0.6.24251 -> 0.7.33634

marsam committed:
Merge pull request #102013 from andir/promscale

promscale: 0.1.0 -> 0.1.1

marsam committed:
Merge pull request #102009 from r-ryantm/auto-update/atlantis

atlantis: 0.15.0 -> 0.15.1

balsoft committed:
Merge pull request #101997 from cab404/super-slicer-2-2-54-2

super-slicer: 2.2.54.1 -> 2.2.54.2

marsam committed:
Merge pull request #102021 from r-ryantm/auto-update/cargo-tarpaulin

cargo-tarpaulin: 0.14.3 -> 0.15.0

marsam committed:
Merge pull request #101841 from helsinki-systems/gif-for-cli-1.1.2

gif-for-cli: unstable-2018-08-14 -> 1.1.2

marsam committed:
Merge pull request #102022 from r-ryantm/auto-update/cfssl

cfssl: 1.4.1 -> 1.5.0

primeos committed:
Merge pull request #101957 from meutraa/update/android-studio-canary

androidStudioPackages.{canary,dev}: 4.2.0.14 -> 4.2.0.15

AndersonTorres committed:
Merge pull request #101247 from wheelsandmetal/add-nuraft-lib

nuraft: init at 1.1.2

unode committed:
lumpy: 0.3.0 -> 0.3.1

r-ryantm committed:
croc: 8.5.2 -> 8.6.5

primeos committed:
google-chrome: Add myself as maintainer

Mainly to get pinged on PRs/issues as well.

r-ryantm committed:
ctop: 0.7.3 -> 0.7.4

primeos committed:
chromium: Drop useVaapi (was deprecated)

peterhoeg committed:
nmapsi4: 0.4.80-20180430 -> 0.5-alpha2

kevincox committed:
Merge pull request #91567 from Vonfry/vimPlugins/vim-airline-clock

vimPlugins: vim-airline-clock init at 2018-05-08

r-ryantm committed:
doctl: 1.48.1 -> 1.49.0

r-ryantm committed:
dstask: 0.22 -> 0.23

r-ryantm committed:
emplace: 0.3.7 -> 0.3.8

marsam committed:
Merge pull request #102040 from r-ryantm/auto-update/ctop

ctop: 0.7.3 -> 0.7.4

r-ryantm committed:
ergo: 3.3.4 -> 3.3.5

r-ryantm committed:
flow: 0.136.0 -> 0.137.0

r-ryantm committed:
flyctl: 0.0.144 -> 0.0.145

matthew-piziak committed:
tdlib: 1.6.0 -> 1.6.9

fix hash

version 1.6.9 is not released, so use revision hash

add a comment to the rev about why we are not using `v${version}` anymore

use unstable-2020-10-25

update version, not revision

unode committed:
pythonPackages.pysam: Disable failing tests unrelated to packaging

These will be fixed upstream and are due to format changes introduced
in samtools and htslib 1.11

domenkozar committed:
servant: generate docs using python3

szlend committed:
maintainers: change szlend email

r-ryantm committed:
fsmon: 1.8.1 -> 1.8.2

domenkozar committed:
Merge pull request #102062 from szlend/szlend-mail-change

Maintainers: change szlend email address

danieldk committed:
Merge pull request #101531 from danieldk/sentencepiece-0.1.94

sentencepiece: 0.1.93 -> 0.1.94

r-ryantm committed:
geoipupdate: 4.3.0 -> 4.5.0

doronbehar committed:
Merge pull request #101747 from doronbehar/pkg/syncthingtray

siraben committed:
mozwire: 0.5.2 -> 0.7.0

risicle committed:
terraform-providers.cloudfoundry: init at 0.12.6 (#99960)

zimbatm committed:
terraform-providers.time: init at 0.6.0 (#101406)

mweinelt committed:
python3Packages.asgi-csrf: disable tests

The packages tests import a private module from httpx, that does not
exist anymore in newer versions.

mweinelt committed:
python3Packages.asgi-csrf: 0.7 -> 0.7.1

r-ryantm committed:
gosec: 2.4.0 -> 2.5.0

rnhmjoj committed:
Merge pull request #102001 from rnhmjoj/pdns

pdns-recursor: 4.3.5 -> 4.4.0

r-ryantm committed:
gotestsum: 0.5.4 -> 0.6.0

lheckemann committed:
Merge pull request #102015 from pstn/murmur-group

nixos/murmur: add murmur group, don't run as nogroup

vbgl committed:
ocamlPackages.curly: unstable-2019-11-14 → 0.2.0

timokau committed:
Merge pull request #100510 from trepetti/noaa-apt-1.3.0

noaa-apt: 1.2.0 -> 1.3.0

timokau committed:
Merge pull request #100653 from trepetti/marker

marker: init at 2020.04.04

r-ryantm committed:
batsignal: 1.1.2 -> 1.1.3

grahamc committed:
nixos ec2/create-amis.sh: shellcheck: quote region references

grahamc committed:
nixos ec2/create-amis.sh: shellcheck: quote state_dir reference

grahamc committed:
nixos ec2/create-amis.sh: shellcheck: SC2155: Declare and assign separately to avoid masking return values.

grahamc committed:
nixos ec2/create-amis.sh: shellcheck: read without -r mangles backslashes

grahamc committed:
nixos ec2/create-amis.sh: shellcheck: explicitly make the additions to block_device_mappings single strings

grahamc committed:
nixos ec2/create-amis.sh: shellcheck: $ is not needed in arithmetic

emptyflask committed:
zoom-us: 5.3.469451.0927 -> 5.4.53350.1027

ryantm committed:
Merge pull request #102045 from r-ryantm/auto-update/doctl

doctl: 1.48.1 -> 1.49.0

ajs124 committed:
Merge pull request #101748 from helsinki-systems/upd/raspberrypifw20201022

raspberrypifw: 1.20200902 -> 1.20201022

Commit:
libguestfs: inherit hydraPlatforms from appliance

(cherry picked from commit 265f2f576f94d7cbc3149304f2d6d7406b6ee8db)

flokli committed:
nixos/networkd: allow RouteMetric= in [DHCPv6] section

r-ryantm committed:
_1password-gui: 0.9.0 -> 0.9.1

doronbehar committed:
Merge pull request #102083 from emptyflask/zoom-us-5.4

zoom-us: 5.3.469451.0927 -> 5.4.53350.1027

evalexpr committed:
vimPlugins.telescope-nvim: init at 2020-10-28

r-ryantm committed:
beancount: 2.3.2 -> 2.3.3

kloenk committed:
openbgpd: 6.7p0 -> 6.8p0

shawndooley committed:
fldigi: 4.1.14 -> 4.1.15

Frostman committed:
blackbox-exporter: 0.17.0 -> 0.18.0

mweinelt committed:
Merge pull request #102096 from flokli/networkd-dhcpv6-route-metric

nixos/networkd: allow RouteMetric= in [DHCPv6] section

dywedir committed:
Merge pull request #101512 from omasanori/nuspell-4.0.0

nuspell: 3.1.2 -> 4.0.0, fix license

Frostman committed:
prometheus-pushgateway: 1.2.0 -> 1.3.0

NeQuissimus committed:
linux: 4.14.202 -> 4.14.203

NeQuissimus committed:
linux: 4.19.152 -> 4.19.153

NeQuissimus committed:
linux: 4.4.240 -> 4.4.241

NeQuissimus committed:
linux: 4.9.240 -> 4.9.241

NeQuissimus committed:
linux: 5.4.72 -> 5.4.73

NeQuissimus committed:
linux: 5.8.16 -> 5.8.17

NeQuissimus committed:
linux: 5.9.1 -> 5.9.2

ryantm committed:
Merge pull request #102047 from r-ryantm/auto-update/dstask

dstask: 0.22 -> 0.23

mmahut committed:
Merge pull request #102051 from r-ryantm/auto-update/ergo

ergo: 3.3.4 -> 3.3.5

r-ryantm committed:
abcl: 1.7.1 -> 1.8.0

kira-bruneau committed:
git-review: build with python3

Also updated source to the new repo and added myself as a maintainer

r-ryantm committed:
consul: 1.8.4 -> 1.8.5

mkenigs committed:
alibuild: use python3

Helps #101964

mweinelt committed:
Merge pull request #102101 from Frostman/prometheus-pushgateway-1.3.0

prometheus-pushgateway: 1.2.0 -> 1.3.0

Ericson2314 committed:
Merge pull request #102069 from nix-windows/libarchive-windows

libarchive: Fix MinGW build

ryantm committed:
Merge pull request #101682 from r-ryantm/auto-update/lynis

lynis: 3.0.0 -> 3.0.1

Ma27 committed:
up: 0.3.2 -> 0.4.0

https://github.com/akavel/up/releases/tag/v0.4

shawndooley committed:
fldigi: 4.1.14 -> 4.1.15
Corrected hash format

lopsided98 committed:
writers.writeNginxConfig: fix cross-compilation

Native versions of awk, sed and gixy need to be used.

marsam committed:
Merge pull request #102055 from r-ryantm/auto-update/flow

flow: 0.136.0 -> 0.137.0

cdepillabout committed:
spago: 0.16.0 -> 0.17.0

r-ryantm committed:
cargo-audit: 0.12.1 -> 0.13.1

ajs124 committed:
attica: drop

as far as I can tell, this file is completely unused.

Everything that uses attica uses it from pkgs/development/libraries/kde-frameworks/attica.nix

r-ryantm committed:
bpytop: 1.0.44 -> 1.0.47

r-ryantm committed:
broot: 1.0.3 -> 1.0.4

bhipple committed:
Merge pull request #102120 from helsinki-systems/drop/attica

attica: drop

bhipple committed:
Merge pull request #100266 from jnetod/textadept-update

textadept: 10.2 -> 10.8

bhipple committed:
Merge pull request #99306 from puckipedia/patch-fatresize

fatresize: 1.0.2 -> 1.1.0

bhipple committed:
Merge pull request #102018 from 1000101/blockbook-frontend

blockbook-frontend: fix&update extraConfig example

delroth committed:
plover.dev: fix Qt version pinning

Issue report: https://github.com/NixOS/nixpkgs/issues/65399#issuecomment-719066888

Similar issues in #98067.

Plover seems to work fine with Qt > 5.14 so this is an easy way to fix
the problem (as opposed to keeping the pinning and making it work with
PyQt).

bhipple committed:
Merge pull request #102097 from shawndooley/fldigi-4.1.15

fldigi: 4.1.14 -> 4.1.15

marsam committed:
tmux: update completions

danieldk committed:
Merge pull request #102125 from r-ryantm/auto-update/broot

broot: 1.0.3 -> 1.0.4

rnhmjoj committed:
Merge pull request #101724 from pickfire/patch-3

fontdir: add ttc to font regex

Commit:
python-modules/coordinates: remove mildlyincompetent from the maintainers list

also removed from maintainers-list.nix, motivation for change is
that he was a friend, and he unfortunately passed away earlier this year.

timokau committed:
Merge pull request #95011 from Atemu/undervolt-pl

undervolt: expose power limits as Nixopts

timokau committed:
Merge pull request #102139 from alex-eyre/master

Remove mildlyincompetent from the maintainers list

danieldk committed:
Merge pull request #101668 from Maxwell-lt/update/mullvad-2020.6

mullvad-vpn: 2020.5 -> 2020.6

cdepillabout committed:
Merge pull request #102117 from cdepillabout/spago-0.17

spago: 0.16.0 -> 0.17.0

Mic92 committed:
Merge pull request #102123 from lopsided98/nginx-conf-cross

writers.writeNginxConfig: fix cross-compilation

deifactor committed:
gajim: strictDeps = false (#102141)

* gajim: strictDeps = false

Without this it doesn't find GSettings schemas, which causes it to crash when (for example) you try to change your avatar.

7c6f434c committed:
Merge pull request #102105 from r-ryantm/auto-update/abcl

abcl: 1.7.1 -> 1.8.0

marsam committed:
tmux: 3.1b -> 3.1c

marsam committed:
Merge pull request #102118 from r-ryantm/auto-update/cargo-audit

cargo-audit: 0.12.1 -> 0.13.1

marsam committed:
Merge pull request #102110 from r-ryantm/auto-update/consul

consul: 1.8.4 -> 1.8.5

marsam committed:
Merge pull request #102080 from r-ryantm/auto-update/gotestsum

gotestsum: 0.5.4 -> 0.6.0

marsam committed:
Merge pull request #102075 from siraben/mozwire-0.7.0

mozwire: 0.5.2 -> 0.7.0

marsam committed:
Merge pull request #102082 from r-ryantm/auto-update/batsignal

batsignal: 1.1.2 -> 1.1.3

marsam committed:
Merge pull request #102079 from r-ryantm/auto-update/gosec

gosec: 2.4.0 -> 2.5.0

marsam committed:
Merge pull request #102074 from r-ryantm/auto-update/geoipupdate

geoipupdate: 4.3.0 -> 4.5.0

marsam committed:
Merge pull request #102065 from r-ryantm/auto-update/fsmon

fsmon: 1.8.1 -> 1.8.2

marsam committed:
Merge pull request #102058 from r-ryantm/auto-update/flyctl

flyctl: 0.0.144 -> 0.0.145

marsam committed:
Merge pull request #102050 from r-ryantm/auto-update/emplace

emplace: 0.3.7 -> 0.3.8

marsam committed:
Merge pull request #102039 from r-ryantm/auto-update/croc

croc: 8.5.2 -> 8.6.5

RaghavSood committed:
ledger-live-desktop: 2.14.0 -> 2.15.0

marsam committed:
Merge pull request #102148 from marsam/update-tmux

tmux: 3.1b -> 3.1c

NeQuissimus committed:
linux: 4.19.153 -> 4.19.154

NeQuissimus committed:
linux_latest-libre: 17724 -> 17744

NeQuissimus committed:
oh-my-zsh: 2020-10-27 -> 2020-10-29

peti committed:
haskell: update Cabal overrides for the new 3.2.1.0 release

WilliButz committed:
grafana: 7.3.0 -> 7.3.1

https://github.com/grafana/grafana/releases/tag/v7.3.1

marsam committed:
Merge pull request #99920 from ericdallo/add-dart-to-flutter

flutter: Bump and add dart cache to flutter

WilliButz committed:
Merge pull request #102094 from Frostman/blackbox-exporter-0.18.0

blackbox-exporter: 0.17.0 -> 0.18.0

mkenigs committed:
broadlink-cli: 0.9.0 -> 0.15.0 and use python3

Helps #101964

grahamc committed:
stage-1: add datestamps to logs

When the stage-1 logs get imported in to the journal, they all get
loaded with the same timestamp. This makes it difficult to identify
what might be taking a long time in early boot.

mkenigs committed:
amazon-glacier-cmd-interface: remove package

Last real upstream activity appears to be ~6 years ago and depends on
python2 packages that will soon be vulnerable

Helps #101964

ajs124 committed:
skype_call_recorder: drop

hasn't worked since 2017: https://atdot.ch/scr/2017/06/important-information-regarding-skype-version-5/
the version packaged here is also from 2008, as far as I can tell
the last upstream release was 2016

grahamc committed:
stage-1: modprobe ext{2,3,4} before resizing

I noticed booting a system with an ext4 root which expanded to 5T took
quite a long time (12 minutes in some cases, 43(!) in others.)

I changed stage-1 to run `resize2fs -d 62` for extra debug output and
timing information. It revealed the adjust_superblock step taking
almost all of the time:

    [Fri Oct 30 11:10:15 UTC 2020] zero_high_bits_in_metadata: Memory used: 132k/0k (63k/70k), time:  0.00/ 0.00/ 0.00
    [Fri Oct 30 11:21:09 UTC 2020] adjust_superblock: Memory used: 396k/4556k (295k/102k), time: 654.21/ 0.59/ 5.13

but when I ran resize2fs on a disk with the identical content growing
to the identical target size, it would only take about 30 seconds. I
looked at what happened between those two steps in the fast case with
strace and found:

```
   235	getrusage(RUSAGE_SELF, {ru_utime={tv_sec=0, tv_usec=1795}, ru_stime={tv_sec=0, tv_usec=3590}, ...}) = 0
   236	write(1, "zero_high_bits_in_metadata: Memo"..., 84zero_high_bits_in_metadata: Memory used: 132k/0k (72k/61k), time:  0.00/ 0.00/ 0.00
   237	) = 84
   238	gettimeofday({tv_sec=1604061278, tv_usec=480147}, NULL) = 0
   239	getrusage(RUSAGE_SELF, {ru_utime={tv_sec=0, tv_usec=1802}, ru_stime={tv_sec=0, tv_usec=3603}, ...}) = 0
   240	gettimeofday({tv_sec=1604061278, tv_usec=480192}, NULL) = 0
   241	mmap(NULL, 2564096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fa3c7355000
   242	access("/sys/fs/ext4/features/lazy_itable_init", F_OK) = 0
   243	brk(0xf85000)                           = 0xf85000
   244	brk(0xfa6000)                           = 0xfa6000
   245	gettimeofday({tv_sec=1604061278, tv_usec=538828}, NULL) = 0
   246	getrusage(RUSAGE_SELF, {ru_utime={tv_sec=0, tv_usec=58720}, ru_stime={tv_sec=0, tv_usec=3603}, ...}) = 0
   247	write(1, "adjust_superblock: Memory used: "..., 79adjust_superblock: Memory used: 396k/2504k (305k/92k), time:  0.06/ 0.06/ 0.00
   248	) = 79
   249	gettimeofday({tv_sec=1604061278, tv_usec=539119}, NULL) = 0
   250	getrusage(RUSAGE_SELF, {ru_utime={tv_sec=0, tv_usec=58812}, ru_stime={tv_sec=0, tv_usec=3603}, ...}) = 0
   251	gettimeofday({tv_sec=1604061279, tv_usec=939}, NULL) = 0
   252	getrusage(RUSAGE_SELF, {ru_utime={tv_sec=0, tv_usec=520411}, ru_stime={tv_sec=0, tv_usec=3603}, ...}) = 0
   253	write(1, "fix_uninit_block_bitmaps 2: Memo"..., 88fix_uninit_block_bitmaps 2: Memory used: 396k/2504k (305k/92k), time:  0.46/ 0.46/ 0.00
   254	) = 88
```

In particular the access to /sys/fs seemed interesting. Looking
at the source of resize2fs:

```
[root@ip-172-31-22-182:~/e2fsprogs-1.45.5]# rg -B2 -A1 /sys/fs/ext4/features/lazy_itable_init .
./resize/resize2fs.c
923-	if (getenv("RESIZE2FS_FORCE_LAZY_ITABLE_INIT") ||
924-	    (!getenv("RESIZE2FS_FORCE_ITABLE_INIT") &&
925:	     access("/sys/fs/ext4/features/lazy_itable_init", F_OK) == 0))
926-		lazy_itable_init = 1;
```

I confirmed /sys is mounted, and then found a bug suggesting the
ext4 module is maybe not loaded:
https://bugzilla.redhat.com/show_bug.cgi?id=1071909

My home server doesn't have ext4 loaded and had 3T to play with, so
I tried (and succeeded with) replicating the issue locally:

```
[root@kif:/scratch]# lsmod | grep -i ext

[root@kif:/scratch]# zfs create -V 3G rpool/scratch/ext4

[root@kif:/scratch]# time mkfs.ext4 /dev/zvol/rpool/scratch/ext4
mke2fs 1.45.5 (07-Jan-2020)
Discarding device blocks: done
Creating filesystem with 786432 4k blocks and 196608 inodes
Filesystem UUID: 560a4a8f-93dc-40cc-97a5-f10049bf801f
Superblock backups stored on blocks:
	32768, 98304, 163840, 229376, 294912

Allocating group tables: done
Writing inode tables: done
Creating journal (16384 blocks): done
Writing superblocks and filesystem accounting information: done

real	0m2.261s
user	0m0.000s
sys	0m0.025s

[root@kif:/scratch]# zfs set volsize=3T rpool/scratch/ext4

[root@kif:/scratch]# time resize2fs -d 62 /dev/zvol/rpool/scratch/ext4
resize2fs 1.45.5 (07-Jan-2020)
fs has 11 inodes, 1 groups required.
fs requires 16390 data blocks.
With 1 group(s), we have 22234 blocks available.
Last group's overhead is 10534
Need 16390 data blocks in last group
Final size of last group is 26924
Estimated blocks needed: 26924
Extents safety margin: 49
Resizing the filesystem on /dev/zvol/rpool/scratch/ext4 to 805306368 (4k) blocks.
read_bitmaps: Memory used: 132k/0k (63k/70k), time:  0.00/ 0.00/ 0.00
read_bitmaps: I/O read: 1MB, write: 0MB, rate: 3802.28MB/s
fix_uninit_block_bitmaps 1: Memory used: 132k/0k (63k/70k), time:  0.00/ 0.00/ 0.00
resize_group_descriptors: Memory used: 132k/0k (68k/65k), time:  0.00/ 0.00/ 0.00
move_bg_metadata: Memory used: 132k/0k (68k/65k), time:  0.00/ 0.00/ 0.00
zero_high_bits_in_metadata: Memory used: 132k/0k (68k/65k), time:  0.00/ 0.00/ 0.00
```

here it got stuck for quite some time ... straceing this 20 minutes in revealed this in a tight loop:

```
getuid()                                = 0
geteuid()                               = 0
getgid()                                = 0
getegid()                               = 0
prctl(PR_GET_DUMPABLE)                  = 1 (SUID_DUMP_USER)
fallocate(3, FALLOC_FL_ZERO_RANGE, 2222649901056, 2097152) = 0
fsync(3)                                = 0
```

it finally ended 43(!) minutes later:

```
adjust_superblock: Memory used: 264k/3592k (210k/55k), time: 2554.03/ 0.16/15.07
fix_uninit_block_bitmaps 2: Memory used: 264k/3592k (210k/55k), time:  0.16/ 0.16/ 0.00
blocks_to_move: Memory used: 264k/3592k (211k/54k), time:  0.00/ 0.00/ 0.00
Number of free blocks: 755396/780023556, Needed: 0
block_mover: Memory used: 264k/3592k (216k/49k), time:  0.05/ 0.01/ 0.00
block_mover: I/O read: 1MB, write: 0MB, rate: 18.68MB/s
inode_scan_and_fix: Memory used: 264k/3592k (216k/49k), time:  0.00/ 0.00/ 0.00
inode_ref_fix: Memory used: 264k/3592k (216k/49k), time:  0.00/ 0.00/ 0.00
move_itables: Memory used: 264k/3592k (216k/49k), time:  0.00/ 0.00/ 0.00
calculate_summary_stats: Memory used: 264k/3592k (216k/49k), time: 16.35/16.35/ 0.00
fix_resize_inode: Memory used: 264k/3592k (222k/43k), time:  0.04/ 0.00/ 0.00
fix_resize_inode: I/O read: 1MB, write: 0MB, rate: 22.80MB/s
fix_sb_journal_backup: Memory used: 264k/3592k (222k/43k), time:  0.00/ 0.00/ 0.00
overall resize2fs: Memory used: 264k/3592k (222k/43k), time: 2570.90/16.68/15.07
overall resize2fs: I/O read: 1MB, write: 1MB, rate: 0.00MB/s
The filesystem on /dev/zvol/rpool/scratch/ext4 is now 805306368 (4k) blocks long.

real	43m1.943s
user	0m16.761s
sys	0m15.069s
```

I then cleaned up and recreated the zvol, loaded the ext4 module, created the ext4 fs,
resized the volume, and resize2fs'd and it went quite quickly:

```
[root@kif:/scratch]# zfs destroy rpool/scratch/ext4

[root@kif:/scratch]# zfs create -V 3G rpool/scratch/ext4

[root@kif:/scratch]# modprobe ext4

[root@kif:/scratch]# time resize2fs -d 62 /dev/zvol/rpool/scratch/ext4

[root@kif:/scratch]# time mkfs.ext4 /dev/zvol/rpool/scratch/ext4
mke2fs 1.45.5 (07-Jan-2020)
Discarding device blocks: done
Creating filesystem with 786432 4k blocks and 196608 inodes
Filesystem UUID: 5b415f2f-a8c4-4ba0-ac1d-78860de77610
Superblock backups stored on blocks:
	32768, 98304, 163840, 229376, 294912

Allocating group tables: done
Writing inode tables: done
Creating journal (16384 blocks): done
Writing superblocks and filesystem accounting information: done

real	0m1.013s
user	0m0.001s
sys	0m0.023s

[root@kif:/scratch]# zfs set volsize=3T rpool/scratch/ext4

[root@kif:/scratch]# time resize2fs -d 62 /dev/zvol/rpool/scratch/ext4
resize2fs 1.45.5 (07-Jan-2020)
fs has 11 inodes, 1 groups required.
fs requires 16390 data blocks.
With 1 group(s), we have 22234 blocks available.
Last group's overhead is 10534
Need 16390 data blocks in last group
Final size of last group is 26924
Estimated blocks needed: 26924
Extents safety margin: 49
Resizing the filesystem on /dev/zvol/rpool/scratch/ext4 to 805306368 (4k) blocks.
read_bitmaps: Memory used: 132k/0k (63k/70k), time:  0.00/ 0.00/ 0.00
read_bitmaps: I/O read: 1MB, write: 0MB, rate: 3389.83MB/s
fix_uninit_block_bitmaps 1: Memory used: 132k/0k (63k/70k), time:  0.00/ 0.00/ 0.00
resize_group_descriptors: Memory used: 132k/0k (68k/65k), time:  0.00/ 0.00/ 0.00
move_bg_metadata: Memory used: 132k/0k (68k/65k), time:  0.00/ 0.00/ 0.00
zero_high_bits_in_metadata: Memory used: 132k/0k (68k/65k), time:  0.00/ 0.00/ 0.00
adjust_superblock: Memory used: 264k/1540k (210k/55k), time:  0.02/ 0.02/ 0.00
fix_uninit_block_bitmaps 2: Memory used: 264k/1540k (210k/55k), time:  0.15/ 0.15/ 0.00
blocks_to_move: Memory used: 264k/1540k (211k/54k), time:  0.00/ 0.00/ 0.00
Number of free blocks: 755396/780023556, Needed: 0
block_mover: Memory used: 264k/3592k (216k/49k), time:  0.01/ 0.01/ 0.00
block_mover: I/O read: 1MB, write: 0MB, rate: 157.11MB/s
inode_scan_and_fix: Memory used: 264k/3592k (216k/49k), time:  0.00/ 0.00/ 0.00
inode_ref_fix: Memory used: 264k/3592k (216k/49k), time:  0.00/ 0.00/ 0.00
move_itables: Memory used: 264k/3592k (216k/49k), time:  0.00/ 0.00/ 0.00

calculate_summary_stats: Memory used: 264k/3592k (216k/49k), time: 16.20/16.20/ 0.00
fix_resize_inode: Memory used: 264k/3592k (222k/43k), time:  0.00/ 0.00/ 0.00
fix_resize_inode: I/O read: 1MB, write: 0MB, rate: 5319.15MB/s
fix_sb_journal_backup: Memory used: 264k/3592k (222k/43k), time:  0.00/ 0.00/ 0.00
overall resize2fs: Memory used: 264k/3592k (222k/43k), time: 16.45/16.38/ 0.00
overall resize2fs: I/O read: 1MB, write: 1MB, rate: 0.06MB/s
The filesystem on /dev/zvol/rpool/scratch/ext4 is now 805306368 (4k) blocks long.

real	0m17.908s
user	0m16.386s
sys	0m0.079s
```

Success!

ajs124 committed:
linuxstopmotion: 0.8.0 -> 0.8.5

and qt4 -> qt5

Mic92 committed:
Merge pull request #101948 from matthew-piziak/tdlib-169

danieldk committed:
Merge pull request #102089 from r-ryantm/auto-update/1password

_1password-gui: 0.9.0 -> 0.9.1

grahamc committed:
create-amis.sh: log the full response if describing the import snapshot tasks fails

Mic92 committed:
Merge pull request #100814 from unode/samtools

grahamc committed:
create-amis: allow customizing the service role name

The complete setup on the AWS end can be configured
with the following Terraform configuration. It generates
a ./credentials.sh which I just copy/pasted in to the
create-amis.sh script near the top. Note: the entire stack
of users and bucket can be destroyed at the end of the
import.

    variable "region" {
      type = string
    }
    variable "availability_zone" {
      type = string
    }

    provider "aws" {
      region = var.region
    }

    resource "aws_s3_bucket" "nixos-amis" {
      bucket_prefix = "nixos-amis-"
      lifecycle_rule {
        enabled = true
        abort_incomplete_multipart_upload_days = 1
        expiration {
          days = 7
        }
      }
    }

    resource "local_file" "credential-file" {
      file_permission = "0700"
      filename = "${path.module}/credentials.sh"
      sensitive_content = <<SCRIPT
    export service_role_name="${aws_iam_role.vmimport.name}"
    export bucket="${aws_s3_bucket.nixos-amis.bucket}"
    export AWS_ACCESS_KEY_ID="${aws_iam_access_key.uploader.id}"
    export AWS_SECRET_ACCESS_KEY="${aws_iam_access_key.uploader.secret}"
    SCRIPT
    }

    # The following resources are for the *uploader*
    resource "aws_iam_user" "uploader" {
      name = "nixos-amis-uploader"
    }

    resource "aws_iam_access_key" "uploader" {
      user = aws_iam_user.uploader.name
    }

    resource "aws_iam_user_policy" "upload-to-nixos-amis" {
      user = aws_iam_user.uploader.name

      policy = data.aws_iam_policy_document.upload-policy-document.json
    }

    data "aws_iam_policy_document" "upload-policy-document" {
      statement {
        effect = "Allow"

        actions = [
          "s3:ListBucket",
          "s3:GetBucketLocation",
        ]

        resources = [
          aws_s3_bucket.nixos-amis.arn
        ]
      }

      statement {
        effect = "Allow"

        actions = [
          "s3:PutObject",
          "s3:GetObject",
          "s3:DeleteObject",
        ]

        resources = [
          "${aws_s3_bucket.nixos-amis.arn}/*"
        ]
      }

      statement {
        effect = "Allow"
        actions = [
          "ec2:ImportSnapshot",
          "ec2:DescribeImportSnapshotTasks",
          "ec2:DescribeImportSnapshotTasks",
          "ec2:RegisterImage",
          "ec2:DescribeImages"
        ]
        resources = [
          "*"
        ]
      }
    }

    # The following resources are for the *vmimport service user*
    # See: https://docs.aws.amazon.com/vm-import/latest/userguide/vmie_prereqs.html#vmimport-role
    resource "aws_iam_role" "vmimport" {
      assume_role_policy = data.aws_iam_policy_document.vmimport-trust.json
    }

    resource "aws_iam_role_policy" "vmimport-access" {
      role = aws_iam_role.vmimport.id
      policy = data.aws_iam_policy_document.vmimport-access.json
    }

    data "aws_iam_policy_document" "vmimport-access" {
      statement {
        effect = "Allow"
        actions = [
          "s3:GetBucketLocation",
          "s3:GetObject",
          "s3:ListBucket",
        ]
        resources = [
          aws_s3_bucket.nixos-amis.arn,
          "${aws_s3_bucket.nixos-amis.arn}/*"
        ]
      }
      statement {
        effect = "Allow"
        actions = [
          "ec2:ModifySnapshotAttribute",
          "ec2:CopySnapshot",
          "ec2:RegisterImage",
          "ec2:Describe*"
        ]
        resources = [
          "*"
        ]
      }
    }

    data "aws_iam_policy_document" "vmimport-trust" {
      statement {
        effect = "Allow"
        principals {
          type = "Service"
          identifiers = [ "vmie.amazonaws.com" ]
        }

        actions = [
          "sts:AssumeRole"
        ]

        condition {
          test = "StringEquals"
          variable = "sts:ExternalId"
          values = [ "vmimport" ]
        }
      }
    }

mmahut committed:
Merge pull request #101847 from mmilata/python-trezor-udev-linux

python3Packages.trezor: make udev rules dependency linux-only

Commit:
motion 4.3.1 -> 4.3.2

veprbl committed:
Merge pull request #102169 from cheriimoya/motion

motion 4.3.1 -> 4.3.2

grahamc committed:
create-amis: improve wording around the service name's IAM role

Co-authored-by: Cole Helbling <cole.e.helbling@outlook.com>

stigtsp committed:
Merge pull request #102066 from zakame/contrib/perl-NetAsyncHTTP

perlPackages.NetAsyncHTTP: init at 0.47

stigtsp committed:
Merge pull request #102071 from zakame/contrib/perl-NetAsyncWebSocket

perlPackages.NetAsyncWebSocket: init at 0.13

grahamc committed:
amazon-image: random.trust_cpu=on to cut 10s from boot

Ubuntu and other distros already have this set via kernel config.

Ma27 committed:
citrix_workspace: remove attributes for old versions; fix i686 build

Ma27 committed:
citrix_workspace: add 20.10.0

ChangeLog: https://docs.citrix.com/en-us/citrix-workspace-app-for-linux/whats-new.html#whats-new-in-2010

Ma27 committed:
Merge pull request #102160 from WilliButz/update/grafana/v7.3.1

grafana: 7.3.0 -> 7.3.1

matthewbauer committed:
Merge pull request #96318 from matthewbauer/provide-patchelf-in-native-stdenv

stdenv/native: provide patchelf on linux

Lassulus committed:
Merge pull request #88544 from Pablo1107/ledger2beancount-2.1

ledger2beancount: init at 2.1

Lassulus committed:
Merge pull request #102002 from RaghavSood/electrum-ltc/ledger

electrum-ltc: enable hardware wallets

ajs124 committed:
Merge pull request #101869 from helsinki-systems/upd/nginx

nginxMainline: 1.19.3 -> 1.19.4

peti committed:
hackage2nix: disable broken builds to fix evaluation

worldofpeace committed:
Merge pull request #101112 from deviant/remove-uemacs

uemacs: remove

worldofpeace committed:
Merge pull request #101067 from deviant/remove-caddy-agree

nixos/caddy: remove services.caddy.agree

grahamc committed:
Merge pull request #102171 from grahamc/faster-ext-resize

stage-1: modprobe ext{2,3,4} before resizing (so resizing takes less than 45 minutes)

grahamc committed:
Merge pull request #102172 from grahamc/stage-1-datestamps

stage-1: add datestamps to logs

grahamc committed:
Merge pull request #102173 from grahamc/create-amis

create-amis.sh: fixup shellcheck issues, improve error logging, and add configurable service names

grahamc committed:
Merge pull request #102175 from grahamc/ami-random

amazon-image: random.trust_cpu=on to cut 10s from boot

grahamc committed:
Merge pull request #102174 from grahamc/ami-root-use-gpt

AMI root partition table: use GPT to support >2T partitions

flyfloh committed:
python-miio: fix

LnL7 committed:
Merge pull request #101783 from elseym/plex

plex: 1.20.2.3402 -> 1.20.3.3483

worldofpeace committed:
Merge pull request #101384 from eyJhb/among-sus

among-sus: init at 2020-10-22

worldofpeace committed:
maintainers: my name should be stylized

jtojnar committed:
Merge pull request #102163 from mkenigs/rm-glacier-cmd

amazon-glacier-cmd-interface: remove package

peti committed:
Merge pull request #101829 from NixOS/haskell-updates

Update Haskell package set to LTS 16.20 (plus other fixes)

worldofpeace committed:
Merge pull request #101132 from deviant/iosevka-variants

iosevka-bin: add support for variants

mweinelt committed:
Merge pull request #99523 from mweinelt/ircdog

ircdog: init at 0.2.1

mweinelt committed:
Merge pull request #99521 from mweinelt/irc-stuff

python3Packages.{irctokens,ircstates,ircrobots}: init

deviant committed:
lib/sources.nix: fix incorrect inherit

split comes from builtins, not lib.

	error: attribute 'split' missing, at /nix/path/nixpkgs/lib/sources.nix:4:4
	(use '--show-trace' to show detailed location information)

SuperSandro2000 committed:
shfmt: 3.1.2 -> 3.2.0

zowoq committed:
buildah: 1.16.5 -> 1.17.0

https://github.com/containers/buildah/releases/tag/v1.17.0

Ma27 committed:
Merge pull request #101421 from sikmir/geogebra

geogebra: 5-0-593-0 -> 5-0-609-0, geogebra6: 6-0-600-0 -> 6-0-609-0

Ma27 committed:
Merge pull request #102121 from r-ryantm/auto-update/bpytop

bpytop: 1.0.44 -> 1.0.47

Ma27 committed:
Merge pull request #101679 from r-ryantm/auto-update/liquibase

liquibase: 4.0.0 -> 4.1.1

infinisil committed:
Merge pull request #102191 from deviant/fix-inherit

lib/sources.nix: fix incorrect inherit

zowoq committed:
shfmt: install man page

ipetkov committed:
topgrade: 5.7.0 -> 5.8.1

jbedo committed:
last: 1066 -> 1133

evalexpr committed:
i3status-rust: 0.14.1 -> 0.14.2

grahamc committed:
Merge pull request #102201 from danderson/danderson/add-cli-to-path

nixos/tailscale: add tailscale to environment.systemPackages.

ajs124 committed:
Merge pull request #102164 from helsinki-systems/drop/skype-call-recorder

skype_call_recorder: drop

bhipple committed:
Merge pull request #102198 from jbedo/last

last: 1066 -> 1133

bhipple committed:
Merge pull request #102197 from ipetkov/topgrade-5.8.1

topgrade: 5.7.0 -> 5.8.1

bhipple committed:
Merge pull request #102183 from flyfloh/miio-fix

python-miio: fix

omasanori committed:
kmscon: add omasanori as a maintainer (#102206)

Signed-off-by: Masanori Ogino <167209+omasanori@users.noreply.github.com>

worldofpeace committed:
Merge pull request #100136 from xaverdh/nixos-install-support-impure

nixos-install: pass through impure flag

lf- committed:
nixos/postgresql: fix inaccurate docs for authentication (#97622)

* nixos/postgresql: fix inaccurate docs for authentication

We actually use peer authentication, then md5 based authentication.
trust is not used.

* Use a link for mkForce docs

Co-authored-by: aszlig <aszlig@redmoonstudios.org>

Co-authored-by: lf- <lf-@users.noreply.github.com>
Co-authored-by: aszlig <aszlig@redmoonstudios.org>

worldofpeace committed:
Merge pull request #101438 from nora-puchreiner/patch-1

graalvm8-ee, graalvm11-ee: 20.0.0 -> 20.2.1

mmahut committed:
Merge pull request #99929 from JesusMtnez/slack

slack: 4.9.0 -> 4.10.3

mmahut committed:
Merge pull request #102152 from RaghavSood/ledger-live/2.15.0

ledger-live-desktop: 2.14.0 -> 2.15.0

Ma27 committed:
Merge pull request #102199 from evalexpr/i3status-rust-update-0.14.2

i3status-rust: 0.14.1 -> 0.14.2

zowoq committed:
vgrep: 2.4.0 -> 2.5.0

https://github.com/vrothberg/vgrep/releases/tag/v2.5.0

SuperSandro2000 committed:
rubocop: 0.77.0 -> 1.1.0

primeos committed:
tdesktop: 2.4.4 -> 2.4.5

ryantm committed:
Merge pull request #90322 from pasqui23/zplg2

zplugin: init at 2.3

Ma27 committed:
packer: 1.6.4 -> 1.6.5

https://github.com/hashicorp/packer/releases/tag/v1.6.5

Ma27 committed:
Merge pull request #102228 from SuperSandro2000/rubocop-1.1.0

rubocop: 0.77.0 -> 1.1.0

Commit:
Merge branch 'master' of github.com:gefla/nixpkgs

SuperSandro2000 commented:
Please fix that PR. Going to close until then.

gefla commented:
Not planning to continue work on this since I've stopped using Lightworks. If anyone wants to pick this up, I'd suggest updating to the most recent upstream version first since Lightworks has progressed beyond 14.5.0 by now.
